### DISCUSSION TITLE
Refactor `org.kiwiproject.jaxrs.exception` package when Jakarta RS transition happens

### DISCUSSION THREAD

**Discussion by chrisrohr** [kiwiproject/kiwi#1095] [General]
We should _not_ do anything about this now, but I _really_ don't like the way we are constructing an exception class which is then  passed as an argument to another method. Whenever JAX-RS finishes the transition to Jakarta RS and all the package names change, we can and should look to re-design everything that is currently in the `org.kiwiproject.jaxrs.exception` package.

_Originally posted by @sleberknight in https://github.com/kiwiproject/dropwizard-service-utilities/pull/37#discussion_r538858410_

**Comment by sleberknight** [comment]
Note that we won't change anything that currently exists in the `org.kiwiproject.jaxrs.exception` package. Instead we will create a new package for Jakarta RS and consider it a blank slate. Some things might migrate more or less intact with only package changes, e.g. `KiwiResponses`, but the `exception` package and things related to it like `KiwiJaxrsValidations` should be re-considered.

**Comment by sleberknight** [comment]
Clearly this did not happen when we transitioned to Jakarta EE. At this point it is a very low priority. In my original comment I did not like creating an exception to pass into another method, which then creates a `Response` from the exception. Perhaps there is a way to take code like this:

```java
return JaxrsExceptionMapper.buildResponse(new JaxrsBadRequestException(exception));
```

and create a new way of building the response without creating another exception, e.g. by simply passing the desired status code along with the original exception:

```java
return JaxrsExceptionMapper.buildResponse(400, exception);
```

And of course there can be overloads, e.g.

```java
return JaxrsExceptionMapper.buildResponse(Response.Status.BAD_REQUEST, exception);
```

At this point, I think making small improvements is a better use of limited time than some wholesale refactoring or redesign, since we have a lot of production code that already uses the existing design. And while not perfect by any means, it's also not that bad and hasn't caused us any problems in the 7-8 years we've been using it like it currently exists.

**Comment by sleberknight** [comment]
Closing this as the original concern is now OBE — the original design has been in production over many years and a wholesale refactor isn't warranted. Small improvements can be tracked as individual issues if and when they come up.
